cmd/nerdctl/build.go

```diff
@@ -318,5 +331,36 @@ func generateBuildctlArgs(cmd *cobra.Command, platform, args []string) (string,
 		buildctlArgs = append(buildctlArgs, "--export-cache="+s)
 	}
 
+	iidFile, err := cmd.Flags().GetString("iidfile")
+	if err != nil {
+		return "", nil, false, cleanup, err
+	}
+	if iidFile != "" {
+		buildctlArgs = append(buildctlArgs, "--metadata-file="+iidFile)
```

Comment:
Confusing to write metadata file to iidfile.
Better to create a metadata file in `os.MkdirTemp`